### PR TITLE
Change upload in createsnapshot to upload subcomponent for pvc

### PR DESCRIPTION
**Pull request by xinyanw409**
Currently when upload snapshot for pvc it will failed due to format error when GetProtectedEntity. This pr change to upload subcomponent for pvc to avoid such issue.

Test ongoing:
https://container-dp.svc.eng.vmware.com/job/Container_Precheck_Velero/262/

Manually tested backup, all upload succeeded.

### CODE REVIEW

File: pkg/snapshotmgr/snapshot_manager.go [hunk 1]
```diff
@@ -261,7 +261,17 @@ func (this *SnapshotManager) createSnapshot(peID astrolabe.ProtectedEntityID, ta
 		return updatedPeID, err
 	}
 
-	uploadBuilder := builder.ForUpload(veleroNs, "upload-"+peSnapID.GetID()).BackupTimestamp(time.Now()).NextRetryTimestamp(time.Now()).SnapshotID(updatedPeID.String()).Phase(v1api.UploadPhaseNew)
+	uploadBuilder := builder.ForUpload(veleroNs, "upload-"+peSnapID.GetID()).BackupTimestamp(time.Now()).NextRetryTimestamp(time.Now()).Phase(v1api.UploadPhaseNew)
+	if peID.GetPeType() == "PersistentVolumeClaim" || peID.GetPeType() == astrolabe.PvcPEType {
```
Comment (lintongj) [inline pkg/snapshotmgr/snapshot_manager.go:265]: peType can only be "pvc", which is `astrolabe.PvcPEType`

File: pkg/snapshotmgr/snapshot_manager.go [hunk 2]
```diff
@@ -261,7 +261,17 @@ func (this *SnapshotManager) createSnapshot(peID astrolabe.ProtectedEntityID, ta
...
+		components, err := pe.GetComponents(ctx)
+		if err != nil {
+			this.WithError(err).Errorf("Failed to retrive subcomponents for %s", peID.String())
+		}
+		subcomponent := components[0].GetID().String() + ":" + peSnapID.GetID()
```
Comment (lintongj) [inline pkg/snapshotmgr/snapshot_manager.go:270]: there is existing API in Astrolabe. Please change to use it.

**Comment by svcbot-qecnsdp** [comment]
Congratulations !! 

All the pre-check tests passed
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/77/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/80/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/81/

File: pkg/snapshotmgr/snapshot_manager.go [hunk 3]
```diff
@@ -261,7 +262,21 @@ func (this *SnapshotManager) createSnapshot(peID astrolabe.ProtectedEntityID, ta
...
+	if peID.GetPeType() == astrolabe.PvcPEType {
+		components, err := pe.GetComponents(ctx)
+		if err != nil {
+			this.WithError(err).Errorf("Failed to retrive subcomponents for %s", peID.String())
+			return updatedPeID, err
+		}
+		if len(components) != 1 {
+			return updatedPeID, errors.New(fmt.Sprintf("Expected 1 component, %s has %d", peID.String(), len(components)))
+		}
+		component := astrolabe.NewProtectedEntityIDWithSnapshotID(components[0].GetID().GetPeType(), components[0].GetID().GetID(), peSnapID)
```
Comment (lintongj) [inline pkg/snapshotmgr/snapshot_manager.go:275]: maybe componentPEID

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/83/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/87/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/99/

**Pull request closed without merging**